staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_work_estimator.go

```diff
@@ -170,18 +171,25 @@ func (e *listWorkEstimator) seatsBasedOnObjectCount(stats storage.Stats, listOpt
 	return uint64(math.Ceil(float64(estimatedObjectsToBeProcessed) / e.config.ObjectsPerSeat))
 }
 
-func (e *listWorkEstimator) seatsBasedOnObjectSize(stats storage.Stats, listOptions metav1.ListOptions, isListFromCache bool) uint64 {
+func (e *listWorkEstimator) seatsBasedOnObjectSize(stats storage.Stats, listOptions metav1.ListOptions, isListFromCache bool, matchesSingle bool) uint64 {
 	// Size not available, fallback to count based estimate
 	if stats.EstimatedAverageObjectSizeBytes <= 0 && stats.ObjectCount != 0 {
-		return e.seatsBasedOnObjectCount(stats, listOptions, isListFromCache)
+		return e.seatsBasedOnObjectCount(stats, listOptions, isListFromCache, matchesSingle)
 	}
 	limited := stats.ObjectCount
 	if listOptions.Limit > 0 && listOptions.Limit < limited {
 		limited = listOptions.Limit
 	}
 	objectsLoadedInMemory := limited
-	if !isListFromCache && (listOptions.FieldSelector != "" || listOptions.LabelSelector != "") {
+	switch {
+	case matchesSingle:
+		objectsLoadedInMemory = 1
+	case isListFromCache:
+		objectsLoadedInMemory = limited
```

Comment:
Linter is complaining about this: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/132950/pull-kubernetes-verify/1945052239789821952

Reply:
Fixed